types/net_trigger.d.ts

```diff
@@ -1,12 +1,12 @@
 import { NetFields } from './net_fields';
 import { NetMatches } from './net_matches';
 
-export class CactbotRegExpExecArray<T extends TriggerTypes> extends RegExpExecArray {
+export interface CactbotRegExpExecArray<T extends TriggerTypes> extends RegExpExecArray {
   groups?: NetMatches[T];
 }
 
-export class CactbotBaseRegExp<T extends TriggerTypes> extends RegExp {
-  exec(string: string): CactbotRegExpExecArray<T> | null;
+export interface CactbotBaseRegExp<T extends TriggerTypes> extends RegExp {
+  exec: (string: string) => CactbotRegExpExecArray<T> | null;
```

Comment:
This looks correct in general, however to account for the `undefined` change in net_matches.d.ts, this adjustment is needed:
```ts
  groups?: T extends 'None' ? never : NetMatches[T];
```

Reply:
Sure